app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -198,6 +198,9 @@ def consumer_create
       sync_task(::Actions::Katello::Host::Register, host, System.new, rhsm_params, content_view_environment)
       host.reload
 
+      parent_host = get_parent_host(request.headers)
+      host.subscription_facet.update_attribute(:registered_through, parent_host)
+
```

Comment:
this likely needs to be done via consumer_activate too, and I'd suggest doing it on facts (and possibly upload_package_profile too), so that if the client changes what capsule its using it'll automatically update